crates/typst-library/src/text/mod.rs

```diff
@@ -766,35 +790,97 @@ impl PlainText for Packed<TextElem> {
 }
 
 /// A lowercased font family like "arial".
-#[derive(Clone, Eq, PartialEq, Hash)]
-pub struct FontFamily(EcoString);
+#[derive(Debug, Clone)]
+pub struct FontFamily {
+    name: EcoString,
+    coverage: Option<Regex>,
+}
+
+impl Hash for FontFamily {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.name.hash(state);
+        self.coverage_str().hash(state);
+    }
+}
+
+impl PartialEq for FontFamily {
+    fn eq(&self, other: &Self) -> bool {
+        self.name == other.name && self.coverage_str() == other.coverage_str()
+    }
+}
 
 impl FontFamily {
     /// Create a named font family variant.
     pub fn new(string: &str) -> Self {
-        Self(string.to_lowercase().into())
+        Self::new_with_coverage(string, None).unwrap()
+    }
+
+    pub fn new_with_coverage(
+        string: &str,
+        coverage: Option<Value>,
```

Comment:
I think we could have a new `enum Covers { LatinInCjk, Regex(Regex) }` type that uses `cast!` to cast from a string or regex without allowing an arbitrary `Value` here.

We could even use that across the board, never converting LatinInCjk to a regex in the first place.